src/pages-and-resources/teams/Settings.jsx

```diff
@@ -0,0 +1,150 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button, Form, Icon } from '@edx/paragon';
+import { Add } from '@edx/paragon/icons';
+import { FieldArray } from 'formik';
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { v4 as uuid } from 'uuid';
+import * as Yup from 'yup';
+import { setupYupExtensions, useAppSetting } from '../../utils';
+import AppSettingsModal from '../app-settings-modal/AppSettingsModal';
+import messages from './messages';
+import TeamSetEditor, { TeamSetTypes, TeamSizes } from './TeamSetEditor';
+
+setupYupExtensions();
+
+function TeamSettings({
+  intl,
+  onClose,
+}) {
+  const [teamsConfiguration, saveSettings] = useAppSetting('teamsConfiguration');
+  const blankNewTopic = {
+    name: '',
+    description: '',
+    type: 'open',
+    maxTeamSize: TeamSizes.DEFAULT,
+    id: null,
+  };
+
+  const handleSettingsSave = (values) => {
+    // For newly-added teams, fill in an id.
+    const topics = values.topics.map(topic => ({
+      id: topic.id || uuid(),
+      name: topic.name,
+      type: topic.type,
+      description: topic.description,
+      max_team_size: topic.maxTeamSize,
+    }));
+    saveSettings({ topics, max_team_size: values.maxTeamSize });
+  };
+
+  return (
+    <AppSettingsModal
+      appId="teams"
+      title={intl.formatMessage(messages.heading)}
+      enableAppHelp={intl.formatMessage(messages.enableTeamsHelp)}
+      enableAppLabel={intl.formatMessage(messages.enableTeamsLabel)}
+      learnMoreText={intl.formatMessage(messages.enableTeamsLink)}
+      onClose={onClose}
+      initialValues={{ ...teamsConfiguration }}
+      validationSchema={{
+        maxTeamSize: Yup.number()
+          .required(intl.formatMessage(messages.maxTeamSizeEmpty))
+          .min(TeamSizes.MIN)
+          .max(
+            TeamSizes.MAX,
+            intl.formatMessage(messages.maxTeamSizeTooHigh, {
+              max: TeamSizes.MAX,
+            }),
+          ),
+        topics: Yup.array().of(
+          Yup.object({
+            id: Yup.string(),
+            name: Yup.string()
+              .required(intl.formatMessage(messages.teamSetFormNameEmpty))
+              .trim(),
+            type: Yup.string().oneOf(Object.values(TeamSetTypes)),
+            description: Yup.string()
+              .required(intl.formatMessage(messages.teamSetFormDescriptionError))
+              .trim(),
+            maxTeamSize: Yup.number()
+              .min(TeamSizes.MIN)
+              .max(
+                TeamSizes.MAX,
+                intl.formatMessage(messages.maxTeamSizeTooHigh, {
+                  max: TeamSizes.MAX,
+                }),
+              )
+              .default(null),
```

Comment:
Global max team size validation is not working when a user tries to submit a team form without touch the max team size.
https://loom.com/share/7769189213cc4e10bd2d2338ee8e281b